include/Mat.h

```diff
@@ -1126,21 +1163,24 @@ class Mat {
 
         if(alloc != NULL){
             DimInfo dInfo(new_ndim, new_dims, new_strides);
-            result.allocator->allocate(&result, result.allocator->userdata, dInfo);
+            result.allocator->allocateMeta(&result, result.allocator->userdata, dInfo);
         }
         else{
```

Comment:
change so that first line of func is "if the allocator is NULL, call the default wrap and return" and remove the need for the else and if  down here